mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs

```diff
@@ -49,6 +49,41 @@ internal static void InstallNativeHelper (INativeCertificateHelper helper)
 				Interlocked.CompareExchange (ref nativeHelper, helper, null);
 		}
 
+		public static X509CertificateImpl InitFromHandle (IntPtr handle)
+		{
+#if MONO_FEATURE_APPLETLS && ONLY_APPLETLS // ONLY_APPLETLS should not support any other option
+			return InitFromHandleApple (handle);
+#elif MONOTOUCH_WATCH
+			throw new PlatformNotSupportedException ();
+#else
+
+#if MONO_FEATURE_APPLETLS // If we support AppleTls, which is the default, and not overriding to legacy
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return InitFromHandleApple (handle);
+#endif
+#if !MOBILE
+			return InitFromHandleCore (handle);
+#elif !MONOTOUCH && !XAMMAC
+			throw new NotSupportedException ();
+#endif
+#endif
+		}
+
+		static X509CertificateImpl Import (byte[] rawData)
+		{
+#if MONO_FEATURE_APPLETLS && ONLY_APPLETLS // ONLY_APPLETLS should not support any other option
+			return ImportApple (rawData);
+#elif MONOTOUCH_WATCH
+			throw new PlatformNotSupportedException ();
+#else
+#if MONO_FEATURE_APPLETLS
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return ImportApple (rawData);
+#endif
+			return ImportCore (rawData);
+#endif
```

Comment:
Question - If you have code that does:

`if (true)
      return FirstThing();
 return SecondThing ();
`

will the compiler / optimizer / JIT remove SecondThing from the method? The reason I wrote things the way I did was to make the ONLY_APPLETLS path only have the apple code. But if our sufficiently smart compiler is smart enough to do the right thing / we don't care I can move to this one.